File: Modules/CustomerManagement/CustomerActivity/CustomerActivityForm.cs
namespace MapPin.CustomerManagement.Forms;

[FormScript("CustomerManagement.CustomerActivity")]
[BasedOnRow(typeof(CustomerActivityRow), CheckNames = true)]
public class CustomerActivityForm
{
    public CustomerActivityType ActivityType { get; set; }
    public DateTime             ActivityDate { get; set; }
    public string               Subject      { get; set; }
    public string               Notes        { get; set; }
}

File: Modules/ServerTypes/CustomerManagement/CustomerActivityForm.ts
import { DateEditor, EnumEditor, initFormType, PrefixedContext, StringEditor, TextAreaEditor } from "@serenity-is/corelib";
import { CustomerActivityType } from "./CustomerActivityType";

export interface CustomerActivityForm {
    ActivityType: EnumEditor;
    ActivityDate: DateEditor;
    Subject: StringEditor;
    Notes: TextAreaEditor;
}

export class CustomerActivityForm extends PrefixedContext {
    static readonly formKey = 'CustomerManagement.CustomerActivity';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CustomerActivityForm.init) {
            CustomerActivityForm.init = true;

            var w0 = EnumEditor;
            var w1 = DateEditor;
            var w2 = StringEditor;
            var w3 = TextAreaEditor;

            initFormType(CustomerActivityForm, [
                'ActivityType', w0,
                'ActivityDate', w1,
                'Subject', w2,
                'Notes', w3
            ]);
        }
    }
}

[CustomerActivityType]; // referenced types